extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -116,14 +123,21 @@ protected Producer<String, String> setKafkaProducer()
   @Override
   public void start()
   {
-    scheduler.schedule(this::sendMetricToKafka, 10, TimeUnit.SECONDS);
-    scheduler.schedule(this::sendAlertToKafka, 10, TimeUnit.SECONDS);
-    if (config.getRequestTopic() != null) {
+    if (config.getEventTypes().contains(KafkaEmitterConfig.EventType.METRICS)) {
+      scheduler.schedule(this::sendMetricToKafka, 10, TimeUnit.SECONDS);
```

Comment:
nit, now that we are repeating this more often, we might want to factor out the time interval as a constant so we only define it once